src/com/homepractice/AbstractionPractice.java

```diff
@@ -0,0 +1,35 @@
+package com.homepractice;
+
+// You can not create object of abstract class..so what is the need of abstract class constructor
+// if abstract class contains 50 fields and also has 100 child class then without abstract class constructor 
+//you have to write initialization in every child class ..reduces the code reusablity
+public abstract class AbstractionPractice {
+
+	 String name;
+	 int age;
+	
+	public AbstractionPractice(String name, int age) {
+		super();
```

Comment:
does super needs to be on a first line of constructor? if yes, why?

Reply:
No, not needed here.it will be called internally anyway.